tests/foreman/api/test_subnet.py

```diff
@@ -0,0 +1,399 @@
+        """Subnet parameter can be deleted
+
+        :id: 972b66ec-d506-4fcb-9786-c62f2f79ac1a
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Delete the above subnet parameter
+
+        :expectedresults: The parameter should be deleted from subnet
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier2
+    def test_positive_delete_subnet_parameter_host_impact(self):
+        """Deleting parameter from subnet component deletes the parameter in
+        host inheriting that subnet
+
+        :id: f5174e00-33cd-4008-8112-9c8190859244
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Create host with the above subnet
+            3. Delete the above parameter from subnet
+            4. List subnet parameters for above host
+
+        :expectedresults:
+
+            1. The parameter should be deleted from host
+            2. The parameter should be deleted from host enc
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier2
+    def test_positive_delete_subnet_overrided__parameter_host_impact(self):
+        """Deleting parameter from subnet component doesnt deletes its
+        overridden parameter in host inheriting that subnet
+
+        :id: e8f4a8e8-64ec-4a0f-aee4-14b6e984b470
+
+        :steps:
+
+            1. Create valid subnet with valid parameter
+            2. Create host with the above subnet
+            3. Override subnet parameter value from host
+            4. Delete the above parameter from subnet
+            5. List host parameters
+
+        :expectedresults:
+
+            1. The parameter should not be deleted from host as it becomes
+                host parameter now
+            2. The parameter should not be deleted from host enc as well
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_list_parameters(self):
+        """Satellite lists all the subnet parameters
+
+        :id: ce86d531-bf6b-45a9-81e3-67e1b3398f76
+
+        :steps:
+
+            1. Create subnet with all the details
+            2. Add two parameters in subnet
+            3. List parameters of subnet
+
+        :expectedresults: The satellite should display all the subnet
+            parameters
+        """
+
+    @stubbed
+    @tier2
+    def test_positive_info_parameter(self):
+        """Satellite returns particular subnet parameter info
+
+        :id: 0952ffa0-4f7f-4f8e-97b9-35bbb3e14848
+
+        :steps:
+
+            1. Create subnet with all the details
+            2. Add parameter in subnet
+            3. Info above subnet parameter
+
+        :expectedresults: The satellite should return subnet parameter info
+        """
+
+    @stubbed
+    @skip_if_bug_open('bugzilla', 1470014)
+    @tier3
+    def test_positive_component_overrides_subnet_parameter(self):
+        """Higher priority component parameter overrides same name subnet
```

Comment:
`Higher priority host component parameter overrides subnet parameter with the same name`